CMakeLists.txt

```diff
@@ -14,6 +16,35 @@ if (NOT DEFINED LLVM_SPIRV_BUILD_EXTERNAL)
   endif(CMAKE_SOURCE_DIR STREQUAL CMAKE_CURRENT_SOURCE_DIR)
 endif (NOT DEFINED LLVM_SPIRV_BUILD_EXTERNAL)
 
+# Download spirv.hpp from the official SPIRV-Headers repository.
+# One can skip this step by manually setting
+# LLVM_EXTERNAL_SPIRV_HEADERS_SOURCE_DIR path.
+if (NOT DEFINED LLVM_EXTERNAL_SPIRV_HEADERS_SOURCE_DIR)
+  if (LLVM_SPIRV_BUILD_EXTERNAL)
+    set(LLVM_EXTERNAL_SPIRV_HEADERS_SOURCE_DIR
+      "${CMAKE_CURRENT_BINARY_DIR}/SPIRV-Headers")
+  else() # In-tree build
+    set(LLVM_EXTERNAL_SPIRV_HEADERS_SOURCE_DIR
+      "${CMAKE_SOURCE_DIR}/projects")
+  endif()
+  message(STATUS "SPIR-V Headers location is not specified. Will try to download
+          spirv.hpp from https://github.com/KhronosGroup/SPIRV-Headers into
+          ${LLVM_EXTERNAL_SPIRV_HEADERS_SOURCE_DIR}")
+  ExternalProject_Add(spirv-headers
+    GIT_REPOSITORY    https://github.com/KhronosGroup/SPIRV-Headers.git
+    GIT_TAG           origin/master
+    SOURCE_DIR        ${LLVM_EXTERNAL_SPIRV_HEADERS_SOURCE_DIR}
+    CONFIGURE_COMMAND ""
+    BUILD_COMMAND     ""
+    INSTALL_COMMAND   ""
```

Comment:
Am I right that there are nothing to build in `SPIRV-Headers` project? Do we need to leave a comment here explaining why all those configure/build/install commands are empty?

BTW, is it possible to remove those options completely?

Reply:
Removing these options results in the build system's attempt to use "default" commands for all these steps, which leads build failures precisely because there is nothing to build/install. I'll add a comment to explain this nuance.